### ISSUE TITLE
OWL-ViT Intended Text Input Format

### ISSUE THREAD

**Issue by alex-bene** [google-research/scenic#888]
There is a small confusion in the sources I looked for regarding the intended text input format for OWL when doing object detection.

Let's say I want to detect a cat and a dog. Which of the below ones would be the recommended way to input those into the model?
1. ["cat", "dog"]
2. ["a photo of a cat", "a photo of a dog"]

(If it makes any difference I'm using huggingface to run OWL, however my understanding is that this would be indifferent for huggingface and this library)

**Comment by mjlm** [comment]
The OWLv1 model was trained with queries like `["a photo of a cat", "a photo of a dog"]` (for the self-training phase of OWLv2, we did not add prompt templates like `a photo of {}`). However, given that this is an open-vocabulary model, it will also provide reasonable outputs with `["cat", "dog"]`.

In practice, I would try the different options and see which one works best for your application.

For the evaluations in the OWLv1 and OWLv2 papers, we actually used [seven different prompt templates](https://github.com/google-research/scenic/blob/main/scenic/projects/owl_vit/preprocessing/label_ops.py#L31-L39) and [averaged the classification logits](https://github.com/google-research/scenic/blob/main/scenic/projects/owl_vit/evaluator.py#L248-L249) over the prompt templates, which improves LVIS AP by about 1.5 points.
